cmd/gitops/beta/run/cmd.go

```diff
@@ -224,40 +226,50 @@ func betaRunCommandRunE(opts *config.Options) func(*cobra.Command, []string) err
 		if fluxVersion, err := run.GetFluxVersion(log, ctx, kubeClient); err != nil {
 			log.Warningf("Flux is not found: %v", err.Error())
 
-			components := flags.Components
-			components = append(components, flags.ComponentsExtra...)
+			product := fluxinstall.NewProduct(flags.FluxVersion)
 
-			if err := run.ValidateComponents(components); err != nil {
-				return fmt.Errorf("can't install flux: %w", err)
-			}
+			installer := fluxinstall.NewInstaller()
 
-			installOpts := install.MakeDefaultOptions()
-			installOpts.Version = flags.FluxVersion
-			installOpts.Namespace = flags.Namespace
-			installOpts.Components = components
-			installOpts.ManifestFile = "flux-system.yaml"
-			installOpts.Timeout = flags.Timeout
+			execPath, err := installer.Ensure(ctx, product)
+			if err != nil {
+				execPath, err = installer.Install(ctx, product)
+				if err != nil {
+					return err
+				}
+			}
 
-			man, err := run.NewManager(log, ctx, kubeClient, kubeConfigArgs)
+			wd, err := os.Getwd()
 			if err != nil {
-				log.Failuref("Error creating resource manager")
 				return err
 			}
 
-			if err := run.InstallFlux(log, ctx, installOpts, man); err != nil {
```

Comment:
:warning: I believe there's at least some code that's now unused - could you please have a go at cleaning it up as well?